src/overlays/actors/ovl_Obj_Hgdoor/z_obj_hgdoor.c

```diff
@@ -21,36 +37,172 @@ const ActorInit Obj_Hgdoor_InitVars = {
+    }
+    this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, header);
+    this->rotation = 0;
+    this->timer = 0;
+    this->cutscene = this->dyna.actor.cutscene;
+    ObjHgdoor_SetupCheckShouldOpen(this);
+}
+
+void ObjHgdoor_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHgdoor* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void ObjHgdoor_SetupCheckShouldOpen(ObjHgdoor* this) {
+    this->actionFunc = ObjHgdoor_CheckShouldOpen;
+}
+
+void ObjHgdoor_CheckShouldOpen(ObjHgdoor* this, GlobalContext* globalCtx) {
+    if (!(gSaveContext.weekEventReg[75] & 0x20) && !(gSaveContext.weekEventReg[52] & 0x20) &&
+        (this->dyna.actor.xzDistToPlayer < 100.0f) && (this->dyna.actor.yDistToPlayer < 40.0f) &&
+        (OBJHGDOOR_IS_RIGHT_DOOR(this))) {
+        ObjHgdoor_SetChild(this, globalCtx);
+        ObjHgdoor_SetParent(this, globalCtx);
+        func_80BD42AC(this);
+    }
+}
+
+void func_80BD42AC(ObjHgdoor* this) {
+    this->actionFunc = func_80BD42C0;
+}
+
+void func_80BD42C0(ObjHgdoor* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->cutscene)) {
+        ActorCutscene_Start(this->cutscene, &this->dyna.actor);
+        func_80BD433C(this);
+        func_80BD433C((ObjHgdoor*)this->dyna.actor.child);
+        return;
+    }
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+    }
+    ActorCutscene_SetIntentToPlay(this->cutscene);
+}
+
+void func_80BD433C(ObjHgdoor* this) {
+    this->unk166 = 0x63;
+    this->actionFunc = func_80BD4358;
+}
+
+void func_80BD4358(ObjHgdoor* this, GlobalContext* globalCtx) {
+    u32 actionIndex;
+
+    if (func_800EE29C(globalCtx, 0x1E3)) {
+        actionIndex = func_800EE200(globalCtx, 0x1E3);
+        if (this->unk166 != globalCtx->csCtx.actorActions[actionIndex]->unk0) {
+            this->unk166 = globalCtx->csCtx.actorActions[actionIndex]->unk0;
+            switch (globalCtx->csCtx.actorActions[actionIndex]->unk0) {
+                case 1:
+                    Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_WOOD_DOOR_OPEN_SPEEDY);
+                    if ((this->dyna.actor.parent != NULL) && (this->dyna.actor.parent->id == ACTOR_EN_HG)) {
+                        this->dyna.actor.parent->colChkInfo.health = 1;
+                    }
+                    D_80BD4690 = 1;
+                    break;
+                case 2:
+                    break;
+            }
+        }
+        if ((D_80BD4690 == 1) && (func_80BD44D0(this, globalCtx))) {
+            func_80BD4460(this);
+        }
+    } else {
+        this->unk166 = 0x63;
+    }
+}
+
+void func_80BD4460(ObjHgdoor* this) {
+    this->timer = 0;
+    this->actionFunc = func_80BD4478;
+}
+
+void func_80BD4478(ObjHgdoor* this, GlobalContext* globalCtx) {
+    if (this->timer++ > 0x50 && !ActorCutscene_GetCanPlayNext(this->cutscene)) {
```

Comment:
```suggestion
    if ((this->timer++ > 80) && !ActorCutscene_GetCanPlayNext(this->cutscene)) {
```
Timers are usually in decimal